src/hooks/useDesktopNotifications.ts

```diff
@@ -0,0 +1,144 @@
+import { useEffect, useState } from 'react';
+import { useConfig } from '../contexts/ConfigProvider';
+import { PrData } from '../types';
+import useLogin from './useLogin';
+
+export default function useDesktopNotifications(rawData: PrData[]) {
+  const login = useLogin();
+  const { notificationsEnabled, notifyForComments } = useConfig().config;
+  const [previousData, setPreviousData] = useState<PrData[]>(rawData);
+
+  useEffect(() => {
+    const notifyOfUpdates = () => {
```

Comment:
I have it in it's own function so we can use the early return while still setting the previous data. The alternative is to wrap it all in an if which could be easier to read. Will give it a go

Reply:
right, I see, I think it would still be cleaner to just put a guard at the top:

```tsx
useEffect(() => {
  if (!login ...) {
    setPreviousData(rawData);
    return;
  }

  // rest of function insides ...

  setPreviousData(rawData);
}, [/* ... */]);
```